src/python/test/test_dxpy.py

```diff
@@ -1877,6 +1877,23 @@ def test_retry_after_without_header_set(self):
         self.assertTrue(50000 <= time_elapsed)
         self.assertTrue(time_elapsed <= 70000)
 
+    @unittest.skip("Example of a request that should be retried")
```

Comment:
Yes, it does take too long, but I also had no way to assert that retries were being made (apart from checking the time taken, which seemed lame to me).

Perhaps part of the way to test this is to generate the exceptions and then add a unit test on `_is_retryable_exception(e)`. I like your idea of controlling the sleep delay in tests too-- actually sleeping to wait for time to elapse in tests doesn't really sit well with me.